js/tests/enhancedDate.spec.js

```diff
@@ -0,0 +1,72 @@
+describe("enhancedDate Test", function () {
+
+    describe("Testing setDate method", function () {
+        it("should be an instance of a Date object", function () {
+            expect(enhancedDate.setDate()).is.an.instanceof(Date);
+        });
+        it("should equal to now", function () {
+            expect(enhancedDate.setDate().getTime()).to.deep.equal(Date.now());
+        });        
+    });
+    
+    describe("Testing getDate method", function() {
+        it("should have type number", function () {
+            expect(enhancedDate.getDate()).to.be.a("number");
+        });
+        it("should be close to NOW in milliseconds", function () {
+            expect(enhancedDate.getDate()).to.be.closeTo(Date.now(), 10);
```

Comment:
getDate() in this test is using a Date that was set previously, so it's unlikely that less than 10 milliseconds have passed before this test runs, meaning this test can frequently fail as well